### ISSUE TITLE
B cannot be cast to org.bouncycastle.asn1.DERSequence

### ISSUE THREAD

**Issue by Kcchouette** [msdhedhi/VerifyWinFileDigitalSignature#2]
Not sure it still maintained but here my error:
```
Exception in thread "main" java.lang.ClassCastException: [B cannot be cast to org.bouncycastle.asn1.DERSequence
	at com.dhedhi.utils.verifywinsign.WinFile.loadFileSecurityInfo(WinFile.java:272)
	at com.dhedhi.utils.verifywinsign.VerifyWinSign.verify(VerifyWinSign.java:85)
```

I call your classes in this way
```
public class MyClass {
    public static void main(String[] args) {
	VerifyWinSign verify = new VerifyWinSign();
	try {
	    boolean a = verify.verify("/home/user/exe.exe");
	    System.out.println("a :" + a);
	} catch (CertificateException | OperatorCreationException | IOException | CMSException e) {
	    // TODO Auto-generated catch block
	    e.printStackTrace();
	}
    }
}
```

https://github.com/msdhedhi/VerifyWinFileDigitalSignature/blob/6bbce6056f2dce3ae1ab25efabc3fb3095c04d19/src/main/java/com/dhedhi/utils/verifywinsign/WinFile.java#L272-L274
from debug
 - signature is CMSSignedData object
 - signature.signedContent is CMSProcessableByteArray object
 - signature.ContentInfo.content says it's BERSequence

**Comment by msdhedhi** [comment]
I cannot seem to find a exe file where the signature is encoded using BERSequence instead of DERSequence. From https://people.eecs.berkeley.edu/~jonah/bc/org/bouncycastle/asn1/BERSequence.html, the BERSequence is suppose to extend a DERSequence. You could try changing the following 2 lines to use a BERSequence
            DERSequence seq = (DERSequence) signature.getSignedContent().getContent();
            DERSequence seq1 = (DERSequence) seq.getObjectAt(1);

OR let me know how I can get hold of your windows signed file and I can try it out.

**Comment by Kcchouette** [comment]
I have fully reproduced using this exe: `atlassian-confluence-7.12.1-x64.exe`

To have it:
 - https://www.atlassian.com/software/confluence/download-archives
 - **Latest release** then choose **Windows 64 Bit**
 - then **agreed**

And test it

Thanks

**Comment by Kcchouette** [comment]
Changing
```
DERSequence seq = (DERSequence) signature.getSignedContent().getContent();
DERSequence seq1 = (DERSequence) seq.getObjectAt(1);
```

to

```
BERSequence seq = (BERSequence) signature.getSignedContent().getContent();
BERSequence seq1 = (BERSequence) seq.getObjectAt(1);
```

give me this errors:

```
Exception in thread "main" java.lang.ClassCastException: [B cannot be cast to org.bouncycastle.asn1.BERSequence
	at com.dhedhi.utils.verifywinsign.WinFile.loadFileSecurityInfo(WinFile.java:274)
	at com.dhedhi.utils.verifywinsign.VerifyWinSign.verify(VerifyWinSign.java:85)
```

**Comment by msdhedhi** [comment]
Thanks for reporting this issue. I have checked in a fix which resolves the issue with "atlassian-confluence-7.12.1-x64.exe". Please try it now and let me know if you still run into any problems.

**Comment by Kcchouette** [comment]
Yes, fixed, closing it
Fixed with commit https://github.com/msdhedhi/VerifyWinFileDigitalSignature/commit/35dcd02089ca0c746cacf688d047559929c77c1b
